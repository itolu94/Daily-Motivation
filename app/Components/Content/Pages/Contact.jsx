import React, {Component} from 'react';
import API from '../../../Utility/API.jsx';


export default class Contact extends Component {
    constructor(){
        super();
        this.state ={
            email: '',
            message: '',
            response: '',
            option: "Select One",
        };
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }
    handleSubmit(e){
        e.preventDefault();
        let {email, message, option} = this.state;
        if(option !== "Select One"){
            let data = {email, message, option};
            API.contact(data, (response) =>{
                if(response.status=== 200) this.setState({response: response.data});
                else this.setState({response: "Oops.. something went wrong.  Please try again later"});
            });
        } else {
            this.setState({response: "Please select an option"});
        }
    }
    handleChange(e){
        this.setState({[e.target.name]: e.target.value});
    }
    render(){
        return(
            <div id='delete-account-div' className="container">
                <div className="row pages-div">
                    <div className="col m8 offset-m2 center-align">
                        <h3 id='delete-account-header'>Please enter your email and message.</h3>
                        <p>{this.state.response}</p>
                        <form onSubmit={this.handleSubmit}>
                            <input
                                type='text'
                                name='email'
                                placeholder='Email Address'
                                value={this.state.email}
                                onChange={this.handleChange}
                                required
                            />
                            <select name="option" value={this.state.option} onChange={this.handleChange}>
                                <option value="Select One">Select One</option>
                                <option value="New Quote">New Quote</option>
                                <option value="Registration Error">Registration Error</option>
                                <option value="Delete Account">Delete Account</option>
                                <option value="Other">Other</option>
                            </select>
                            <textarea
                                name='message'
                                row="10"
                                columns="30"
                                maxLength="200"
                                placeholder='Enter your message'
                                value={this.state.message}
                                onChange={this.handleChange}
                                required
                            />
                            <input className='' type="submit"/>
                        </form>
                    </div>
                </div>
            </div>
        )
    }
}