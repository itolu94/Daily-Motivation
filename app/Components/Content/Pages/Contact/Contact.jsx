import React, {Component} from 'react';
import API from '../../../../Utility/API.jsx';
import {ContactDiv, ContactHeader} from './Contact.style.jsx';

export default class Contact extends Component {
    constructor(){
        super();
        this.state ={
            email: '',
            message: '',
            response: '',
            option: "Select One",
        };
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }
    handleSubmit(e){
        e.preventDefault();
        let {email, message, option} = this.state;
        if(option !== "Select One"){
            let data = {email, message, option};
            API.contact(data, (response) =>{
                if(response.status=== 200) this.setState({response: response.data});
                else this.setState({response: "Oops.. something went wrong.  Please try again later"});
            });
        } else {
            this.setState({response: "Please select an option"});
        }
    }
    handleChange(e){
        this.setState({[e.target.name]: e.target.value});
    }
    render(){
        return(
            <ContactDiv className="container">
                <div className="row pages-div">
                    <div className="col m8 offset-m2 center-align">
                        <ContactHeader id='delete-account-header'>Please enter your email and message.</ContactHeader>
                        <p>{this.state.response}</p>
                        <form onSubmit={this.handleSubmit}>
                            <input
                                type='text'
                                name='email'
                                placeholder='Email Address'
                                value={this.state.email}
                                onChange={this.handleChange}
                                required
                            />
                            <select name="option" value={this.state.option} onChange={this.handleChange}>
                                <option value="Select One">Select One</option>
                                <option value="New Quote">New Quote</option>
                                <option value="Registration Error">Registration Error</option>
                                <option value="Delete Account">Delete Account</option>
                                <option value="Other">Other</option>
                            </select>
                            <textarea
                                name='message'
                                row="10"
                                columns="30"
                                maxLength="200"
                                placeholder='Enter your message'
                                value={this.state.message}
                                onChange={this.handleChange}
                                required
                            />
                            <input className='' type="submit"/>
                        </form>
                    </div>
                </div>
            </ContactDiv>
        )
    }
}