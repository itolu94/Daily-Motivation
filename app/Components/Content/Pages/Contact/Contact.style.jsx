import styled from 'styled-components';

const ContactDiv = styled.div`
    background-color: powderblue;
    color: black;
    margin-top: 10%;
    padding: 10px 0;
`;

const ContactHeader = styled.h3`
    font-size: 1.92rem;
    margin:0;
    padding:0;
    height:100%;
`;

export {ContactDiv, ContactHeader};
