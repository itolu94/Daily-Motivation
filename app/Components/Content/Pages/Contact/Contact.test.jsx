import React from 'react';
import { shallow } from 'enzyme'
import Contact from './Contact.jsx';


test("Header component", () => {
    const wrapper = shallow(<Contact/>);
    expect(wrapper).toMatchSnapshot();
});
