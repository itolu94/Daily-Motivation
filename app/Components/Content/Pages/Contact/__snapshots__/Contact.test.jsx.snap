// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component 1`] = `
<div
  className="container"
  id="delete-account-div"
>
  <div
    className="row pages-div"
  >
    <div
      className="col m8 offset-m2 center-align"
    >
      <h3
        id="delete-account-header"
      >
        Please enter your email and message.
      </h3>
      <p />
      <form
        onSubmit={[Function]}
      >
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="text"
          value=""
        />
        <select
          name="option"
          onChange={[Function]}
          value="Select One"
        >
          <option
            value="Select One"
          >
            Select One
          </option>
          <option
            value="New Quote"
          >
            New Quote
          </option>
          <option
            value="Registration Error"
          >
            Registration Error
          </option>
          <option
            value="Delete Account"
          >
            Delete Account
          </option>
          <option
            value="Other"
          >
            Other
          </option>
        </select>
        <textarea
          columns="30"
          maxLength="200"
          name="message"
          onChange={[Function]}
          placeholder="Enter your message"
          required={true}
          row="10"
          value=""
        />
        <input
          className=""
          type="submit"
        />
      </form>
    </div>
  </div>
</div>
`;
